### PR TITLE
[KYUUBI #5546][AUTHZ] Reorgnize the package names for rules

### LINKED ISSUE #5546: [Subtask] Refactor the code path of authz
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Refactor the code path of AUTHZ for supporting making more clear to be customized 

### Parent issue

#5401

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To close #5546 
Refactor the code path of authz

- Rule is not specify for ranger, extract to a independent path `rule`
- Collect each. rule's class together
- Move RangerConfigProvider to `ranger` package

The output is 
截屏2023-10-27 下午2 46 53

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
It's a start of #5401 cc @yaooqinn @bowenliang123

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/RuleAuthorization.scala [hunk 1]
```diff
@@ -27,8 +27,9 @@ import org.apache.spark.sql.catalyst.trees.TreeNodeTag
 
 import org.apache.kyuubi.plugin.spark.authz._
 import org.apache.kyuubi.plugin.spark.authz.ObjectType._
-import org.apache.kyuubi.plugin.spark.authz.ranger.RuleAuthorization._
+import org.apache.kyuubi.plugin.spark.authz.ranger.{AccessRequest, AccessResource, AccessType, SparkRangerAuditHandler}
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/RuleAuthorization.scala:30]: Here will be handled in next step

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleEliminatePermanentViewMarker.scala [hunk 2]
```diff
@@ -15,16 +15,16 @@
  * limitations under the License.
  */
 
-package org.apache.kyuubi.plugin.spark.authz.util
+package org.apache.kyuubi.plugin.spark.authz.rule.permanetview
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleEliminatePermanentViewMarker.scala:18]:
```suggestion
package org.apache.kyuubi.plugin.spark.authz.rule
```

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleEliminatePermanentViewMarker.scala]: Done

**Comment by yaooqinn** [comment]
Can we reorg with `rule.{config, auth, datamask, rowfilter}`, these sub dirs are named via functionality. 
- For `showobjects`, we can move them to `rowfilter`, it's a special case
- For those unable to be classified, simply put them in `rule`

**Comment by AngersZhuuuu** [comment]
> Can we reorg with `rule.{config, auth, datamask, rowfilter}`, these sub dirs are named via functionality.
> 
> * For `showobjects`, we can move them to `rowfilter`, it's a special case
> * For those unable to be classified, simply put them in `rule`

Done
截屏2023-10-27 下午2 46 53

File: docs/security/authorization/spark/overview.rst [hunk 3]
```diff
@@ -71,7 +71,7 @@ End-users can disable the AuthZ plugin by modifying Spark's configuration. For e
 
 .. code-block:: sql
 
-   set spark.sql.optimizer.excludedRules=org.apache.kyuubi.plugin.spark.authz.ranger.RuleAuthorization
+   set spark.sql.optimizer.excludedRules=org.apache.kyuubi.plugin.spark.authz.rule.RuleAuthorization
```
Comment (yaooqinn) [inline docs/security/authorization/spark/overview.rst:74]: This is a breaking change, let's avoid it.

Comment (AngersZhuuuu) [reply docs/security/authorization/spark/overview.rst]: DOne

**Review by bowenliang123 — APPROVED** [review]
LGTM.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleApplyPermanentViewMarker.scala [hunk 4]
```diff
@@ -15,21 +15,20 @@
  * limitations under the License.
  */
 
-package org.apache.kyuubi.plugin.spark.authz.ranger
+package org.apache.kyuubi.plugin.spark.authz.rule.permanetview
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleApplyPermanentViewMarker.scala:18]: permanent

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/permanetview/RuleApplyPermanentViewMarker.scala]: done

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master

**Comment by pan3793** [comment]
@AngersZhuuuu This PR has logic conflicts with master branch, reverted to recover CI, please re-submit against the latest master branch

**Pull request closed without merging**